truffle/src/com.oracle.truffle.dsl.processor/src/com/oracle/truffle/dsl/processor/parser/NodeParser.java

```diff
@@ -1576,6 +1576,7 @@ private void initializeSpecializations(List<? extends Element> elements, final N
         initializeUninitialized(node);
         initializeOrder(node);
         initializePolymorphism(node); // requires specializations
+        initializeUncached(node);
```

Comment:
initializeUncached uses SpecializationData.getReplaces indirectly that gets initialized in resolveReplaces later.

How about we merge initializeUncached with resolveReplaces. And rename SpecializationData#uncached to Specialization#replaced. This probably needs to be a flattened set again in order to fix the failing test.

Reply:
Done, indeed makes the code simpler :)

Reply:
Perfect. Thanks!